### PR TITLE
Add translate functionality for comments section

### PR DESCRIPTION
**Pull request by claudiachua**
**Used Google Cloud Translation API to translate comments to different languages.** 

**1. pom.xml**
- add dependency to use google translation api libraries

**2. index.html**
- add a drop down list for user to select their preferred language
- 4 options: English, Chinese, French, German
- when user selected from drop down list, it will trigger getTranslatedComment() 

**3. style.css**
- add label tag to have the same style as p tag

**4. script.js**
- add new function: getTranslatedComment()
- add user-selected language to end of the query string before fetching
- remove previous list elements in #comment-container & append with translated comments

**5. DataServlet**
- modify doGet() to get user-selected language (if selected) and translate it

End Product can be viewed here:
https://pchua-sps-summer20.et.r.appspot.com

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -29,27 +29,52 @@
...
+        //Get translation service
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
+
         for (Entity entity : results.asIterable()) {
             String comment = (String) entity.getProperty("comment");
+
+            //Check if foreign language is selected
+            if (languageCode != null) { 
+                Translation translation = translate.translate
+                (comment,
```
Comment (kai-dream) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:64]: Nitpick, the indentation is unusual. Usually if we put multiple function parameters in several lines, we leave the opening parentheses at the end of the first line, so it's a good indication that the first line is not finished yet. And we will indent the next few lines to highlight the relationship between lines.

It usually should look like this:

         Translation translation = translate.translate(
             comment,
             Translate.TranslateOption.targetLanguage(languageCode),
             Translate.TranslateOption.format("text"));

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -8,49 +8,57 @@
...
+        <ul>
+            <li>Heading to CMU this fall to complete my degree and start on my post-grad, but Covid...</li>
+            <li>Retirement dream is to work at Disneyland and sell churros :)</li>
+        </ul>
       
-      <h2>Find Me</h2>
-      <p>If you are interested, this is my <a href="https://www.linkedin.com/in/claudia-chua-pei-si-629b72114/">LinkedIn</a> </p>
-      
-      <h2>If you are bored</h2>
+        <h2>Find Me</h2>
```
Comment (kai-dream) [inline portfolio/src/main/webapp/index.html:28]: I guess those are probably accidental indentation changes, are they?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -50,3 +49,31 @@ function getComment(){
...
+    //Adding Language Code to Query String
+    query = '/data' + '?language='+language ;
+    
+    //GetComments
+    fetch(query).then(response => response.json()).then(comments => { 
+
+        const commentElement = document.getElementById("comment-container");
+        
+        //Removing previous list elements in div container
+        while (commentElement.hasChildNodes()){
```
Comment (kai-dream) [inline portfolio/src/main/webapp/script.js:66]: I think you can also use` commentElement.internalHTML = "" ;` to clear existing comments. It's safer and faster than clearing one by one.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (claudiachua) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:64]: Noted thanks for the tip!

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (claudiachua) [reply portfolio/src/main/webapp/index.html:28]: Yup. I will fix them in my next commit! Thanks for spotting it out

**Pull request merged**